integration/metrics_server_test.go

```diff
@@ -0,0 +1,65 @@
+package integration_test
+
```

Comment:
Just a thought: Since we're now using the `kube-rbac-proxy` provided by kubebuilder as a relay server, do we still want to have these integration tests?

We're basically testing code / components/ APIs that are neither owned by us nor a part of this codebase (FWIW, `kube-rbac-proxy` and the controller-runtime metrics server are probably already being tested in their upstream project repos).

Such integration tests take a lot of time to complete, and prow's flakiness makes things all the worse. IMO if these integration tests aren't a good value add, we should get rid of them.

Reply:
Still gotta test and confirm the reachability of it depending on the provided certs and configuration

Reply:
> Just a thought: Since we're now using kube-rbac-proxy provided by kubebuilder as a relay server, do we still want to have these integration tests? We're basically testing code / components/ APIs that are neither owned by us nor a part of this codebase (FWIW, kube-rbac-proxy and the controller-runtime metrics server are probably already being tested in their upstream project repos).

I agree with the point around not testing controller-runtime metrics server because we aren't even configuring it and it's being directly bootstrapped by kubebuilder. On the other hand, the relay-server is something which we're the ones who are configuring it with all sorts of certs and other config, and Service which we are configuring to expose it. For example, only upon testing, I encountered some configurational issue due to some extra authz which kube-auth-proxy was performing bts resulting in unexpected responses.

At the end of the day, let's realise that our integration tests are actually e2e tests w.r.t addon-operator and the metrics which we're serving are being configured by us only behind the scenes looking at the current state. Hence, from my perspective these should be e2e-tested as well.

> Such integration tests take a lot of time to complete, and prow's flakiness makes things all the worse for our CI. IMO if these integration tests aren't a good value add, we should get rid of them. 

fwiw, the integration test associated with metrics relay server is anyways pretty simple and straightforward enough to barely cause any concerns, so imo shouldn't be a problem to have it.

Reply:
>  we're the ones who are configuring it with all sorts of certs and other config

Makes sense, let's keep these tests then 👍🏻

Reply:
We should not remove any tests here, because we are testing if the interop of all components is working as expected.
Who knows if a new version of that proxy is 100% compatible or we make a config mistake somewhere, breaking functionality.

If we see flaky tests, we need to fix the test - maybe it's racy and we really have a bug or Prow is just slow.
Maintaining these e2e tests is a pain in the a**, but the only alternative is to test every release by hand which is even less fun.

The integration test suite in this repo is the last line of defense before a bug hits prod, so we should be very very careful if we want to exclude something.